File: 2025-02-February-GFG-POTD/February 13 - Pair Sum in BST/Solution.py
class Node:
    def __init__(self, data):
        self.data = data
        self.left = None
        self.right = None

def in_order_traversal(root, nodes):
    if not root:
        return
    in_order_traversal(root.left, nodes)
    nodes.append(root.data)
    in_order_traversal(root.right, nodes)

def find_target(root, target):
    nodes = []
    in_order_traversal(root, nodes)
    left, right = 0, len(nodes) - 1
    while left < right:
        total = nodes[left] + nodes[right]
        if total == target:
            return True
        elif total < target:
            left += 1
        else:
            right -= 1
    return False

File: 2025-02-February-GFG-POTD/February 13 - Pair Sum in BST/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

struct Node {
    int data;
    Node* left;
    Node* right;
    Node(int d) : data(d), left(NULL), right(NULL) {}
};

void inOrderTraversal(Node* root, vector<int>& nodes) {
    if (!root) return;
    inOrderTraversal(root->left, nodes);
    nodes.push_back(root->data);
    inOrderTraversal(root->right, nodes);
}

bool findTarget(Node* root, int target) {
    vector<int> nodes;
    inOrderTraversal(root, nodes);
    int left = 0, right = nodes.size() - 1;
    while (left < right) {
        int sum = nodes[left] + nodes[right];
        if (sum == target) return true;
        else if (sum < target) left++;
        else right--;
    }
    return false;
}
